workbench-core/base/src/services/metadataService.ts

```diff
@@ -44,4 +63,124 @@ export class MetadataService {
       dependencyMetadata
     };
   }
+
+  /**
+   *
+   * @param mainEntityResourceType - main entity resource type
+   * @param mainEntityId - main entity id
+   * @param dependencyResourceType - dependency resource type
+   * @param parser - Zod parser for metadata object
+   * @param queryParams - parameters to query metadata. Supported page size and pagination token.
+   * @returns object containind list of metadata objects and continuation token.
+   */
+  public async listDependentMetadata<DependencyMetadata extends { pk: string; sk: string; id: string }>(
+    mainEntityResourceType: string,
+    mainEntityId: string,
+    dependencyResourceType: string,
+    parser: ZodTypeAny,
+    queryParams?: {
+      pageSize?: number;
+      paginationToken?: string;
+    }
+  ): Promise<{ data: DependencyMetadata[]; paginationToken: string | undefined }> {
+    let params: QueryParams = {
+      key: { name: 'pk', value: `${mainEntityResourceType}#${mainEntityId}` },
+      sortKey: 'sk',
+      begins: { S: `${dependencyResourceType}#` },
+      limit: Math.min(queryParams?.pageSize ?? DEFAULT_API_PAGE_SIZE, MAX_API_PAGE_SIZE)
+    };
+
+    params = addPaginationToken(queryParams?.paginationToken, params);
+
+    const result: QueryCommandOutput = await this._ddbService.query(params).execute();
```

Comment:
You can use [getPaginatedItems](https://github.com/aws-solutions/solution-spark-on-aws/blob/cf576c1e124194171d88a44779899db61b1fbffd/workbench-core/base/src/aws/helpers/dynamoDB/dynamoDBService.ts#L159)